### PR TITLE
Add exchange backend

### PR DESCRIPTION
**Pull request by GuillaumeDSM**
image

### CODE REVIEW

File: octobot_trading/exchanges/exchange_websocket_factory.py [hunk 1]
```diff
@@ -44,10 +45,19 @@ async def _create_websocket(exchange_manager, websocket_class_name, socket_manag
     try:
         exchange_manager.exchange_web_socket = socket_manager.get_websocket_client(exchange_manager.config,
                                                                                    exchange_manager)
+        if exchange_manager.exchange_backend is not None and exchange_manager.exchange.authenticated():
```
Comment (Herklos) [inline octobot_trading/exchanges/exchange_websocket_factory.py:48]: Why here ? I don't think this is the proper place to check if an authenticated account is valid : it's the websocket client creation.

Comment (GuillaumeDSM) [reply octobot_trading/exchanges/exchange_websocket_factory.py:48]: That's why it's a WIP, so that we figure out the best way to do this check

Comment (Herklos) [reply octobot_trading/exchanges/exchange_websocket_factory.py:48]: :+1:

File: octobot_trading/exchanges/exchange_factory.py [hunk 2]
```diff
@@ -69,6 +72,24 @@ async def initialize_real_exchange(exchange_manager):
...
+            exchange_manager.exchange
+        )
+    except Exception as e:
+        exchange_manager.logger.exception(e, True, f"Error when creating exchange backend: {e}")
+
+
+async def _initialize_exchange_backend(exchange_manager):
+    if exchange_manager.exchange_backend is not None and exchange_manager.exchange.authenticated():
+        exchange_manager.is_valid_account, message = await exchange_manager.exchange_backend.is_valid_account()
+        exchange_manager.is_valid_account = True    # TODO remove this line when ready
```
Comment (GuillaumeDSM) [inline octobot_trading/exchanges/exchange_factory.py:87]: TODO

**Review by Herklos — APPROVED** [review]
:+1:

**Review by Herklos — APPROVED** [review]
👍

File: octobot_trading/exchanges/connectors/ccxt_exchange.py [hunk 3]
```diff
@@ -174,10 +174,9 @@ async def get_balance(self, **kwargs: dict):
             balance.pop(enums.ExchangeConstantsCCXTColumns.TIMESTAMP.value, None)
             return balance
 
-        except ccxt.InvalidNonce as e:
-            self.logger.error(f"Error when loading {self.name} real trader portfolio: {e}. "
-                              f"To fix this, please synchronize your computer's clock. ")
-            raise e
+        except ccxt.InvalidNonce as err:
+            exchanges.log_time_sync_error(self.logger, self.name, err, "real trader portfolio")
```
Comment (Herklos) [inline octobot_trading/exchanges/connectors/ccxt_exchange.py:178]: :+1:

File: octobot_trading/exchanges/util/exchange_util.py [hunk 4]
```diff
@@ -73,3 +73,17 @@ def get_order_side(order_type):
...
+
+
+def log_time_sync_error(logger, exchange_name, error, details):
+    docs_url = "https://docs.octobot.online"
+    try:
+        import octobot.constants
+        docs_url = octobot.constants.OCTOBOT_DOCS_URL
+    except ImportError:
+        pass
+    logger.error(
```
Comment (Herklos) [inline octobot_trading/exchanges/util/exchange_util.py:85]: :+1:

**Review by Herklos — APPROVED** [review]

**Pull request merged**